src/parser/SparqlParser.cpp

```diff
@@ -416,12 +416,20 @@ void SparqlParser::parseSolutionModifiers(const string& str,
         i++;
       }
     }
+    if (tokens[i] == "HAVING" && i < tokens.size() - 1) {
+      while (i + 1 < tokens.size() && tokens[i + 1] != "LIMIT" &&
+             tokens[i + 1] != "OFFSET" && tokens[i + 1] != "TEXTLIMIT" &&
+             tokens[i + 1] != "GROUP" && tokens[i + 1] != "ORDER") {
+        addFilter(tokens[i + 1], &query._havingClauses);
+        i++;
+      }
+    }
   }
 }
 
 // _____________________________________________________________________________
 void SparqlParser::addFilter(const string& str,
-                             ParsedQuery::GraphPattern* pattern) {
+                             vector<SparqlFilter>* _filters) {
```

Comment:
I think it might be better if he fixes this according to his needs, as I'm currently not sure what he needs the pattern for. I'd probably either add a flag indicating whether the filter is a HAVING or a normal filter, or simply pass the GraphPattern along as well, but which one is better depends on his specific requirements.